src/collection-preferences/utils.tsx

```diff
@@ -133,6 +137,47 @@ export const ContentDensityPreference = ({ label, description, value, onChange }
   </InternalCheckbox>
 );
 
+interface StickyColumnsPreferenceProps extends CollectionPreferencesProps.StickyColumnsPreference {
+  onChange: (value: any) => void;
+  value?: any;
+}
+
+export const StickyColumnsPreference = ({
+  startColumns,
+  endColumns,
+  onChange,
+  value,
+}: StickyColumnsPreferenceProps) => {
+  return (
+    <InternalSpaceBetween className={styles['sticky-columns']} size="l">
+      {[
+        { side: 'start', preference: startColumns },
+        { side: 'end', preference: endColumns },
+      ].map(item => {
```

Comment:
I think it is a little over-engineering to have an array loop for two options. I would go with this

```js
<InternalSpaceBetween className={styles['sticky-columns']} size="l">
   {startColumns && <StickyPreference preference={startColumns} value={item.start} onChange={(newValue) => onChange({ ...value, start: newValue })} />}
   {endColumns && <StickyPreference preference={endColumns} value={item.end} onChange={(newValue) => onChange({ ...value, end: newValue })} />}
</InternalSpaceBetween>
```

Much easier to read and process, IMO